### DISCUSSION TITLE
Ability to search in environment logs & group by grain name

### DISCUSSION THREAD

**Discussion by TomerAdmon** [QualiTorque/qualitorque.github.io#265] [Ideas]
Hi @shirabendor-quali, 
Developing a new blueprint with multiple technologies and grains will usually require browsing many deployment logs, debugging, updating, and repeating.
To ease the debugging process - I would suggest the following:
- Add a search box for the logs pane allowing to search for specific steps/keywords which will filter the logs only for the relevant events.
- Ability to group logs by grain - that way it will be easier to focus on a specific phase of the orchestration

Thanks!
Tomer

**Comment by shirabendor-quali** [comment]
Thanks @TomerAdmon,
If you click on a specific grain you will get only the logs of that grain. Does this help?

**Reply by TomerAdmon** [reply]
Hi @shirabendor-quali,
The ability to get the logs in a grain context will be awsome!
I think that free search over all logs in an environment will be powerful as well as it will allow highlighting data passing between the grains during orchestration
